ui/ts/components/navbar.ts

```diff
@@ -29,18 +29,22 @@ module Components {
       isActive: (t: Target) => boolean;
     }
 
-    export function controller(ts: TargetSet): any {}
+    export function controller(): any {}
 
-    export function view(ctrl: any, ts: TargetSet): _mithril.MithrilVirtualElement {
-      return m("ul.navigation-sidebar", _.map(ts.targets(), (t: Target) =>
+    export enum NavbarOrientation {Vertical, Horizontal};
+
+    export function view(ctrl: any, args: {ts: TargetSet; orientation: NavbarOrientation; }): _mithril.MithrilVirtualElement {
+      return m(
+        (args.orientation === NavbarOrientation.Vertical ? "ul.navigation-sidebar" : "ul.navigation"),
```

Comment:
Orientation is probably the wrong thing to pass in, but I do like being explicit.